DNN Platform/Library/Entities/Urls/AdvancedUrlRewriter.cs

```diff
@@ -1554,8 +1558,10 @@ private static bool CheckForDebug(HttpRequest request, NameValueCollection query
                     //check ssl enforced
                     if (portalSettings.SSLEnforced)
                     {
+                        var isSecureConnectionOnUnsecurePage = !portalSettings.ActiveTab.IsSecure && result.IsSecureConnection;
+                        
                         //check page is not secure, connection is secure
-                        if (!portalSettings.ActiveTab.IsSecure && result.IsSecureConnection)
+                        if (isSecureConnectionOnUnsecurePage && !isPopUpFromSecurePage)

```

Comment:
Is there any other way to prevent this check from being added to all requests?  A higher level change possibly?

Reply:
@mitchelsellers, probably it's possible if we plan an accurate refactoring. Actually that part of the project seems to be quite redundant and easy to corrupt. I would focus on the easiest way to fix the bug here. Any deep refactoring of the core is welcome but needs time and a big effort.

Reply:
To optimize this, instead of initializing `isPopUpFromSecurePage` every time at the top, we can remove that declaration and also the `isSecureConnectionOnUnsecurePage` declaration, and just do:
```c#
if (result.IsSecureConnection && !portalSettings.ActiveTab.IsSecure && !UrlUtils.IsPopUp(url))
```
This way we call `UrlUtils.IsPopUp` only when necessary.

Reply:
@daguiler Yes, same for me. Just preferred initialization for easy code readability.